File: Misc. Algos/Non_recursive_algos/uniquearray.c
// Finding whether the array has all distinct elements or not
// code by Akshat Aryan
#include <stdio.h>

int unique(int arr[], int n){
    for(int i = 0; i < n - 1; i++){
        for(int j = 0; j < n; j++){
            if(arr[i] == arr[j]){
                return 0; // 0 = False
            }
        }
    }
    return 1; // 1= True
}

int main(){
    int n;
    printf("Enter the number of elements in the array:");
    scanf("%d", &n);
    int arr[n];
    printf("Enter the %d elements separated by spaces:", n);
    for(int i = 0; i < n; i++){
        scanf("%d", &arr[i]);
    }
    if(unique(arr, n) == 1){
        printf("All Array Elements are unique.\n");
    }
    else{
        printf("All Array Elements are not unique.\n");
    }
    return 0;
}

File: Misc. Algos/Non_recursive_algos/uniquearray.py
# Finding whether the array has all distinct elements or not
#code by Akshat Aryan

#mentioned algorithm:
def unique(arr):
    for i in range(0, len(arr) - 1):
        for j in range(i + 1, len(arr)):
            if(arr[i] == arr[j]):
                return False
    return True

#another algorithm possible in python
def unique_set(arr):
    #set only contains distinct elements
    #if number of elements in the set and array are same, means array has unique elements
    if(len(arr) == len(set(arr))):
        return True
    else:
        return False

if(__name__ == "__main__"):
    arr = list(map(int, input("Enter the elements separated by space:").split()))
    print("Using Levitin's Algo, uniqueness =", unique(arr))
    print("Using the other Algo, uniqueness =", unique_set(arr))